react/ProductSpecificationText.test.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react'
+import { render } from '@vtex/test-tools/react'
+import { useProduct, ProductContext } from 'vtex.product-context'
+
+import ProductSpecificationGroup from './ProductSpecificationGroup'
+import ProductSpecification from './ProductSpecification'
+import ProductSpecificationValues from './ProductSpecificationValues'
+import ProductSpecificationText from './ProductSpecificationText'
+
+const mockUseProduct = useProduct as jest.Mock<ProductContext>
+
+test('ProductSpecificationText should render only the context it has available', () => {
+  mockUseProduct.mockImplementation(() => ({
+    product: {
+      specificationGroups: [
+        {
+          name: 'Group',
+          originalName: 'Group',
+          specifications: [
+            {
+              name: 'Specification 1',
+              originalName: 'Specification 1',
+              values: ['Value 1'],
+            },
+          ],
+        },
+      ],
+    },
+  }))
+
+  const { getByText, queryByText, rerender } = render(
+    <ProductSpecificationGroup>
+      <ProductSpecificationText message="{groupName} {specificationName} {specificationValue}" />
+    </ProductSpecificationGroup>
+  )
+
+  expect(getByText('Group')).toBeInTheDocument()
+  expect(queryByText('Specification 1')).not.toBeInTheDocument()
+  expect(queryByText('Value 1')).not.toBeInTheDocument()
+
+  rerender(
+    <ProductSpecificationGroup>
+      <ProductSpecification>
+        <ProductSpecificationText message="{groupName} {specificationName} {specificationValue}" />
+      </ProductSpecification>
+    </ProductSpecificationGroup>
+  )
```

Comment:
Would it make sense to move these to a utility method? Tests would be a bit leaner